tests/test_elements/test_mol2mol/test_mol2mol_filter.py

```diff
@@ -14,68 +22,143 @@
 SMILES_CL_BR = "NC(Cl)(Br)C(=O)O"
 SMILES_METAL_AU = "OC[C@H]1OC(S[Au])[C@H](O)[C@@H](O)[C@@H]1O"
 
+SMILES_LIST = [
+    SMILES_ANTIMONY,
+    SMILES_BENZENE,
+    SMILES_CHLOROBENZENE,
+    SMILES_METAL_AU,
+    SMILES_CL_BR,
+]
+
 
 class MolFilterTest(unittest.TestCase):
     """Unittest for MolFilter, which invalidate molecules based on criteria defined in the respective filter."""
 
     def test_element_filter(self) -> None:
-        """Test if molecules are filtered correctly by allowed chemical elements.
-
-        Returns
-        -------
-        None
-        """
-        smiles2mol = SmilesToMol()
-        default_atoms = {
-            1,
-            5,
-            6,
-            7,
-            8,
-            9,
-            14,
-            15,
-            16,
-            17,
-            34,
-            35,
-            53,
+        """Test if molecules are filtered correctly by allowed chemical elements."""
+        default_atoms_dict = {
+            1: (1, None),
+            5: (1, None),
+            6: (1, None),
+            7: (1, None),
+            8: (1, None),
+            9: (1, None),
+            14: (1, None),
+            15: (1, None),
+            16: (1, None),
+            17: (1, None),
+            34: (1, None),
+            35: (1, None),
+            53: (1, None),
```

Comment:
It seems only atoms present in the mol are checked if they are in the boundary.   
So at the moment it is not possible to enforce that the mol has a certain count of atoms, if it is absent. This might be unexpected behavior.   
Maybe set tuple in default to (0, None).
This would mean Atom can be present, but is not required to be present.  
Then, implicitly all atoms not given in the list are treated as (0,0).  
What do you think?

Reply:
You are generally right. However, in the current logic, I only go through the elements that are in the molecule and then check the allowed_el...list. By this, 1 or 0 in this position does just not matter. But it is true, that if I now set e.g. a range (3,5) for O, and O does not occur in the compount, it will still pass. So I need to adapt the logic here and then apply your suggestion

Reply:
But still, I need to check whether all elements from the mol are actually in the allowed elements list.

Also (0, None) and (None, None) seems to be identical for me. What would you prefer?

Reply:
I think we should allow both and set the default to 0.

Reply:
Good. Also realized that we need to add Hs if 1 is in the allowed_elements_numbers ...

Reply:
Now it should behave as expected